test/char_matcher_test.dart

```diff
@@ -990,5 +990,18 @@ void main() {
               onNonMatch: (nonMatch) => '?$nonMatch?'),
           '?a?!1!?b?!2!?c?');
     });
+    test('astral plane', () {
+      expect('🌐'.startsWith(pattern), isFalse);
+      expect('🌐'.indexOf(pattern), -1);
+      expect('🌐'.lastIndexOf(pattern), -1);
```

Comment:
These are the only two String methods that fail without the fix, but I tested all String methods that accept a Pattern just to be thorough.